astropy/coordinates/baseframe.py

```diff
@@ -276,6 +276,66 @@ def convert_input(self, value):
         return out, converted
 
 
+class CartesianRepresentationFrameAttribute(FrameAttribute):
+    """
+    A frame attribute that is a CartesianRepresentation with specified units.
+
+    Parameters
+    ----------
+    default : object
+        Default value for the attribute if not provided
+    secondary_attribute : str
+        Name of a secondary instance attribute which supplies the value if
+        ``default is None`` and no value was supplied during initialization.
+    unit : unit object or None
+        Name of a unit that the input will be converted into. If None, no
+        unit-checking or conversion is performed
+    """
+    def __init__(self, default=None, secondary_attribute='', unit=None, shape=None):
+        super(CartesianRepresentationFrameAttribute, self).__init__(default, secondary_attribute)
+        self.unit = unit
+
+    def convert_input(self, value):
+        """
+        Checks that the input is a CartesianRepresentation with the correct unit, or
+        the special value ``0``.
+
+        Parameters
+        ----------
+        value : object
+            Input value to be converted.
+
+        Returns
+        -------
+        out, converted : correctly-typed object, boolean
+            Tuple consisting of the correctly-typed object and a boolean which
+            indicates if conversion was actually performed.
+
+        Raises
+        ------
+        ValueError
+            If the input is not valid for this attribute.
+        """
+        if np.all(value == 0) and self.unit is not None:
```

Comment:
Just wondering about this "special value".  It seems like it makes code slightly more difficult to read.  If I saw `obsgeoloc = 0` then I would be confused because `0` is not a valid cartesian point.  If the `0` just to stop from having to type `[0, 0, 0] * u.m`, then perhaps explicit is better here?